includes/classes/order.php

```diff
@@ -832,16 +842,11 @@ function setTaxRatesForProduct($products, $loop, $index, $taxCountryId, $taxZone
     }
 
     /**
-     * Calculate taxes for a specific product and add them to the order
+     * Calculate taxes for a specific product and add them to the order's sub-total
+     * and running-sub-total array for the final tax calculations.
      */
-    function calculateTaxForProduct($index, $taxRates, $taxCountryId, $taxZoneId)
+    protected function calculateTaxForProduct($index, $taxRates)
     {
-        $this->use_external_tax_handler_only = false;
-        $this->notify('NOTIFY_ORDER_CART_EXTERNAL_TAX_HANDLING', [], $index, $taxCountryId, $taxZoneId);
-        if ($this->use_external_tax_handler_only == true) {
-            return;
-        }
```

Comment:
Removal of the `NOTIFY_ORDER_CART_EXTERNAL_TAX_HANDLING` notifier prevents the integration of 3rd party tax-authority tools which could handle all the tax calculations for the order, such as applying per-county rules etc.

Reply:
@scottcwilson checked all currently-posted plugins (and I checked on client's sites that have customized tax-handlers); none of those integrations made use of this notification.  For my clients, I've made use of the following notifications instead:
```
            /* From /includes/classes/order.php, during the cart->order conversion process */
            'NOTIFY_ORDER_CART_AFTER_ADDRESSES_SET',
            'NOTIFY_ORDER_CART_ADD_PRODUCT_LIST',

            /* From /includes/classes/order.php, after creating the base 'orders' entry */
            'NOTIFY_ORDER_DURING_CREATE_ADDED_ORDER_HEADER',

            /* From various functions in /includes/functions/functions_taxes.php */
            'NOTIFY_ZEN_GET_TAX_RATE_OVERRIDE',
            'NOTIFY_ZEN_GET_TAX_DESCRIPTION_OVERRIDE',
            'NOTIFY_ZEN_GET_MULTIPLE_TAX_RATES_OVERRIDE',

            /* From /includes/modules/order_total/ot_shipping.php */
            'NOTIFY_OT_SHIPPING_TAX_CALCS',
```

Reply:
I think it was TaxCloud or something like that which first used it. But they've not cooperated with any invitations to collaborate, for ages. So we can drop it for now.